backend/frontend/views.py

```diff
@@ -72,6 +76,51 @@ def is_super(user):
     return user.is_superuser
 
 
+def user_passes_test_or_forbidden(test_func):
+    """
+    Decorator for views that checks that the user passes the given test,
+    returning a 403 Forbidden response if necessary. The default user_passes_test
+    decorator redirects to the login page, which is not what we want for the
+    frontend, or at least some of the frontend views.
+    """
+
+    def decorator(view_func):
+        @wraps(view_func)
+        def _wrapped_view(request, *args, **kwargs):
+            if test_func(request.user):
+                return view_func(request, *args, **kwargs)
+
+            return HttpResponseForbidden(
```

Comment:
If the user is logged in then they should get a forbidden message. If they are not logged in they should be directed to the login page.